File: site_backend/src/services/gemini.ts
import { GoogleGenAI } from '@google/genai';

export type ResumeMode = 'MODE_A' | 'MODE_B';

export function getGeminiClient() {
  const apiKey = process.env.GEMINI_API_KEY || process.env.API_KEY;
  if (!apiKey) throw new Error('Missing GEMINI_API_KEY');
  return new GoogleGenAI({ apiKey });
}

export function getModelName(): string {
  return process.env.GEMINI_MODEL || 'gemini-3-flash-preview';
}

File: services/app/services/gemini.py
from __future__ import annotations
from google import genai
from google.genai import types
from app.core.config import settings

def get_gemini_client() -> genai.Client:
    if not settings.gemini_api_key:
        raise RuntimeError("Missing GEMINI_API_KEY")
    return genai.Client(api_key=settings.gemini_api_key)

def get_model_name() -> str:
    return settings.gemini_model

def make_config(system_instruction: str) -> types.GenerateContentConfig:
    return types.GenerateContentConfig(system_instruction=system_instruction, temperature=settings.gemini_temperature)
